contents/ruby/Rails.md

```diff
@@ -0,0 +1,154 @@
+                          DELETE /users/:id(.:format)                                                                     users#destroy
+```
+
+As can be seen, the common routes that you would need for the resources have already been mapped to URI's. Rails will then look for a controller called `users_controller.rb`, as is convention, and invoke the name of the function mapped to the particular URI and Verb. More information about Rails routes and how they work can be found [here](https://guides.rubyonrails.org/routing.html#resource-routing-the-rails-default).
+
+#### Controller
+
+In the standard Rails convention, once inside the application directory, if you navigate to `app/controllers/`, you should be able to see all the controllers in your project. Rails will look for the controller with the `<name>_controller.rb`, where `<name>` would be replaced by the result under the `Controller` heading in the list of routes above, in this case, "users". It will then look for the function name under the `Action` heading mapped to the URI function inside the Controller, and then call it. This is where the bulk of the business logic of your application would be. So if the person makes a GET request to `/users`, then Rails will open `users_controller.rb`, and call its `index` function.
+
+#### Model
+
+The Model (the M in MVC) is very similar to a Class (à la Object-Oriented Programming), and is a useful abstraction for representing and encapsulating ideas. In Rails, it is also referred to as the "Active Record", as it follows the Active Record pattern. In this pattern, the Models are mapped to tables in the database, and you can query for the fields in the database directly via the model in the native language (i.e. Ruby), instead of making queries the database's language (SQL, etc).Models also have generic functions such as `find`, `all`, `create`, `save`, `update`, `delete` included in them by default, so you don't need to implement them or write long queries for each. 
+
+In our diagram, the Controller calls the `all` method in the `User` model, which is, by convention in a file called `User.rb`, which is stored along with all other models at `app/models/`. The model then acts as an intermediary between the application and the database, and returns the results of the corresponding query. 
+
+#### View
+
+After the Controller has fetched all the data necessary and applied whatever transformations are needed, it passes in the requisite fields to its view. The View (the V in MVC) is what the client sees, and is a collection of .erb (Embedded RuBy) files, which are basically just HTML files with lines of Ruby code embedded in it to modify its appearance and behaviour. When the controller is generated using the Rails CLI, a corresponding folder of views is made inside `app/views/`, which is used to store all the .erb files, and that is where Rails will look for files to serve up to the client at the end of the request-response cycle. 
```

Comment:
> When the controller is generated using the Rails CLI, a corresponding folder of views is made inside `app/views/`, which is used to store all the .erb files, and that is where Rails will look for files to serve up to the client at the end of the request-response cycle. 

This sentence feels a little long. Perhaps you could split them into multiple sentences?